tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -106,6 +108,11 @@ sealed trait NativeConfig {
   def withEmbedResources(
       value: Boolean
   ): NativeConfig
+
+  def withNativeSourcePlugins(plugins: Seq[NativeSourcePlugin]): NativeConfig
+
+  def withNativeSourcePlugin(plugin: NativeSourcePlugin): NativeConfig =
+    withNativeSourcePlugins(nativeSourcePlugins :+ plugin)
```

Comment:
Actually, I wrote these with different purpose.

`withNativeSourcePlugin` _appends_ a single plugin to your list of plugins.

`withNativeSourcePlugins` _replaces_ the entire list of plugins. Otherwise there is no way to remove and/or change the order.

I have some more thoughts on this topic, in the other comments.